packages/webdriver-utils/src/providers/automateProvider.js

```diff
@@ -0,0 +1,19 @@
+import GenericProvider from './genericProvider.js';
+
+export default class AutomateProvider extends GenericProvider {
+  static supports(commandExecutorUrl) {
+    return commandExecutorUrl.includes(process.env.AA_DOMAIN || 'browserstack');
+  }
+
+  async browserstackExecutor(action, args) {
+    if (!this.driver) throw new Error('Driver is null, please initialize driver with createDriver().');
+    let options = args ? { action, arguments: args } : { action };
+    let res = await this.driver.executeScript({ script: `browserstack_executor: ${JSON.stringify(options)}`, args: [] });
+    return res;
+  }
+
+  async setDebugUrl() {
+    const sessionDetails = await this.browserstackExecutor('getSessionDetails');
```

Comment:
this was an old implementation and took over 1 sec to resolve. We have moved to a better implementation, have a look

Reply:
I get this part here https://github.com/percy/percy-appium-js/blob/master/percy/providers/appAutomateProvider.js#L146-L153, however our seleniumHub changes are not in place. So for v1 revision may be we can leave it but will definitely need to pick in v2.

Reply:
- Instead of `getSessionDetails` we will be using `percySnapshotStart` like we do in app-percy.
- Will need to edit `SeleniumHub` to support `desktop` urls